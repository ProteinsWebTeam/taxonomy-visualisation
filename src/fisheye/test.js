import applyFishEye from '.';

describe('FishEye', () => {
  const tree = {
    x: 100,
    y: 100,
    children: [{ x: 101, y: 99 }, { x: 101, y: 100 }, { x: 101, y: 101 }],
  };
  let global = {};
  let treeCopy = {};

  beforeEach(() => {
    treeCopy = JSON.parse(JSON.stringify(tree));
    global = {
      enableFisheye: true,
      root: treeCopy,
      focused: treeCopy.children[1],
    };
  });

  test('if disabled, the tree does not change', () => {
    global.enableFisheye = false;
    applyFishEye(global);
    expect(tree).toEqual(treeCopy);
  });
  test('if enable, the tree changes', () => {
    applyFishEye(global);
    expect(tree).not.toEqual(treeCopy);
    expect(treeCopy).toMatchSnapshot();
  });
  test('if focused is root, the tree does not changes', () => {
    global.focused = global.root;
    applyFishEye(global);
    expect(tree).toEqual(treeCopy);
  });
});
