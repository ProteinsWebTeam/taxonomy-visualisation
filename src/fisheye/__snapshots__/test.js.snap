// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FishEye if enable, the tree changes 1`] = `
Object {
  "children": Array [
    Object {
      "x": 101,
      "y": 80,
    },
    Object {
      "x": 101,
      "y": 100,
    },
    Object {
      "x": 101,
      "y": 120,
    },
  ],
  "x": 86,
  "y": 100,
}
`;
